### ISSUE TITLE
using blur filter with 0 doesn't return anything instead of sending 400

### ISSUE THREAD

**Issue by bagipriyank** [thumbor/thumbor#1051]
### Thumbor request URL
http://localhost:8000/unsafe/filters:blur(0)/myimage.jpg

### Expected behaviour
Should return 400 Bad Request

### Actual behaviour
returns with gateway timeout 504
Stack trace:
```
2018-03-13 00:05:07 tornado.application:ERROR Future exception was never retrieved: Traceback (most recent call last):
  File "/usr/local/lib/python2.7/site-packages/tornado/gen.py", line 307, in wrapper
    yielded = next(result)
  File "/usr/local/lib/python2.7/site-packages/thumbor/transformer.py", line 138, in smart_detect
    self.do_image_operations()
  File "/usr/local/lib/python2.7/site-packages/thumbor/transformer.py", line 231, in do_image_operations
    callback=inner
  File "/usr/local/lib/python2.7/site-packages/thumbor/context.py", line 287, in queue
    self._execute_in_foreground(operation, callback)
  File "/usr/local/lib/python2.7/site-packages/thumbor/context.py", line 275, in _execute_in_foreground
    callback(result)
  File "/usr/local/lib/python2.7/site-packages/thumbor/transformer.py", line 227, in inner
    self.done_callback()
  File "/usr/local/lib/python2.7/site-packages/thumbor/handlers/__init__.py", line 247, in after_transform
    self.filters_runner.apply_filters(thumbor.filters.PHASE_POST_TRANSFORM, finish_callback)
  File "/usr/local/lib/python2.7/site-packages/thumbor/filters/__init__.py", line 91, in apply_filters
    exec_one_filter()
  File "/usr/local/lib/python2.7/site-packages/thumbor/filters/__init__.py", line 90, in exec_one_filter
    f.run(exec_one_filter)
  File "/usr/local/lib/python2.7/site-packages/thumbor/filters/__init__.py", line 198, in run
    results.append(self.runnable_method(*self.params))
  File "/usr/local/lib/python2.7/site-packages/thumbor/filters/__init__.py", line 23, in wrapper
    return fn(self, *args2)
  File "/usr/local/lib/python2.7/site-packages/thumbor/filters/blur.py", line 43, in blur
    matrix, matrix_size = self.generate_1d_matrix(sigma, radius)
  File "/usr/local/lib/python2.7/site-packages/thumbor/filters/blur.py", line 33, in generate_1d_matrix
    exp = math.e ** -(((adj_x * adj_x)) / two_sigma_squared)
ZeroDivisionError: float division by zero
```

### Operating system
OSx

### Your thumbor.conf
```
TC_AWS_LOADER_BUCKET='mybucket'
LOADER='tc_aws.loaders.presigning_loader'
TC_AWS_REGION='us-east-1'
TC_AWS_ENABLE_HTTP_LOADER='True'
RESULT_STORAGE_STORES_UNSAFE='False' -e STORES_CRYPTO_KEY_FOR_EACH_IMAGE='False'
TC_AWS_STORE_METADATA='False'
OPTIMIZERS="['thumbor.optimizers.jpegtran']"
MAX_AGE='604800'
QUALITY='100'
```

Also tried with opencv_engine and still got the same issue.

**Comment by caeugusmao** [comment]
@bagipriyank 

We reproduced this issue here. Could you send a PR for us with this fix? 

If we are assuming that the min value is 1, even when values less than 1 were passed, we have to warn this in our wiki, could send a PR with this update.

https://github.com/thumbor/thumbor/wiki/Blur

Tks!!!

**Comment by heynemann** [comment]
It would be better to discard blur if the value is not valid, instead of 400. Thumbor usually tries its best to not fail if it is possible to continue. Usually with filters this means ignoring the filter.

**Comment by caeugusmao** [comment]
He is proposing we adopt "1" as min value. In his fork, he had this approach.
https://github.com/jinesh90/thumbor/commit/1ad5a427bbbbe0c2c6e0cdcee305264e006d718a 

This way we would not have response status code equals 400 and Thumbor could keep continue its flow.

**Comment by bagipriyank** [comment]
there is already one pr https://github.com/thumbor/thumbor/pull/1052 for this.

**Comment by caeugusmao** [comment]
@bagipriyank Did you see my comments in your pr #1052 ?

**Comment by bagipriyank** [comment]
it's not me who submitted that pr :)

**Comment by jinesh90** [comment]
folks, sorry for late attention, i will fix pr and resubmit again.

**Comment by caeugusmao** [comment]
Fixed in this PR: #1082

**Comment by jinesh90** [comment]
Thank you @caeugusmao !!! Muito obrigado!!!
